static/js/invite.js

```diff
@@ -56,62 +56,77 @@ exports.launch = function () {
     prepare_form_to_be_shown();
     invitee_emails.focus().autosize();
 
-    $("#invite_user_form").ajaxForm({
-        dataType: 'json',
-        beforeSubmit: function () {
-            reset_error_messages();
-            // TODO: You could alternatively parse the textarea here, and return errors to
-            // the user if they don't match certain constraints (i.e. not real email addresses,
-            // aren't in the right domain, etc.)
-            //
-            // OR, you could just let the server do it. Probably my temptation.
-            $('#submit-invitation').button('loading');
-            return true;
-        },
-        success: function () {
-            $('#submit-invitation').button('reset');
-            invite_status.text(i18n.t('User(s) invited successfully.'))
-                .addClass('alert-success')
-                .show();
-            invitee_emails.val('');
-
-            if (page_params.development_environment) {
-                var rendered_email_msg = templates.render('dev_env_email_access');
-                $('#dev_env_msg').html(rendered_email_msg).addClass('alert-info').show();
-            }
-
-        },
-        error: function (xhr) {
-            $('#submit-invitation').button('reset');
-            var arr = JSON.parse(xhr.responseText);
-            if (arr.errors === undefined) {
-                // There was a fatal error, no partial processing occurred.
-                invite_status.text(arr.msg)
-                    .addClass('alert-error')
-                    .show();
-            } else {
-                // Some users were not invited.
-                var invitee_emails_errored = [];
-                var error_list = $('<ul>');
-                _.each(arr.errors, function (value) {
-                    error_list.append($('<li>').text(value.join(': ')));
-                    invitee_emails_errored.push(value[0]);
-                });
-
-                invite_status.addClass('alert-warning')
-                    .empty()
-                    .append($('<p>').text(arr.msg))
-                    .append(error_list)
+    $("#submit-invitation").on("click", function () {
+        var data = {
+            invitee_emails: $("#invitee_emails").val(),
+            invite_as_admin: $("#invite_as_admin input:checked").val(),
+            csrfmiddlewaretoken: $('input[name="csrfmiddlewaretoken"]').attr('value'),
+        };
+        var streams = [];
+        $.each($("#invite-stream-checkboxes input:checked"), function () {
+            streams.push($(this).val());
+        });
+        data.streams = streams;
```

Comment:
The reason `streams` was transformed to `streams[]` is because jquery appends `[]` after the key of arrays to differentiate them from primitive values. Using `stream_names = request.POST.getlist('streams[]')` should work. Or pass `traditional: true,` to channel.post. (Both will work). @timabbott What do you think is the correct approach here?